features/support/check/checkCookieContents.js

```diff
@@ -0,0 +1,29 @@
+const assert = require('assert').strict;
+
+/**
+ * Checks if an cookie does, or does not, exist on the page. 
+ * Throws an error if the cookie name is not defined in the document.
+ * @param {String} cname Cookie name to check for.
+ * @param {String} cvalue Cookie value to check for.
+ * @param {String} not The string "not" to negate the check (cookie value should not match).
+ */
+module.exports = async function(cname, cvalue, not) {
+
+  var cookieValue; 
+  var cookieNameFound;
+  const pageCookies = await this.page.cookies();
+  const expectedValue = cvalue === undefined ? '' : cvalue;
+  const shouldValueBeEqual = not ? false : true;
+  
+  for (let index = 0; index < pageCookies.length; index++) {
```

Comment:
The way you've coded this is good, but another way you might see this is to initialize a `length` variable at the same time as the `index` so you don't need to keep checking the array's length property:
```javascript
for (let index = 0, length = pageCookies.length; index < length; index++) {
```